src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,39 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final String DETAILS_FOR = "Details for ";
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> findUserData(userId), executor);
+    }
+
+    private UserData findUserData(String userId) {
+        if (cache.containsKey(userId)) {
+            return cache.get(userId);
+        }
+        UserData userData = findUserDataFromDB(userId);
+        cache.put(userId, userData);
```

Comment:
You're blocking here to wait until response from `findUserDataFromDB`, so it's better to separate blocking operation `findUserDataFromDB` from saving user data to cache. Like supplyAsync(blocking operation).whenComplete(put into cache)